Mathlib/CategoryTheory/Limits/Shapes/RegularMono.lean

```diff
@@ -57,6 +59,53 @@ attribute [reassoc] RegularMono.w
 instance (priority := 100) RegularMono.mono (f : X ⟶ Y) [RegularMono f] : Mono f :=
   mono_of_isLimit_fork RegularMono.isLimit
 
+/-- Every isomorphism is a regular monomorphism. -/
+def RegularMono.ofIso (e : X ≅ Y) : RegularMono e.hom where
+  Z := Y
+  left := 𝟙 Y
+  right := 𝟙 Y
+  isLimit := Fork.IsLimit.mk _ (fun s ↦ s.ι ≫ e.inv) (by simp) fun s m w ↦ by
+    dsimp; rw [← w]; simp
+
+/-- Regular monomorphisms are preserved by isomorphisms in the arrow category. -/
+def RegularMono.ofArrowIso {X'} {Y'} {f : X ⟶ Y} {g : X' ⟶ Y'}
+    (e : Arrow.mk f ≅ Arrow.mk g) [h : RegularMono f] :
+    RegularMono g where
+  Z := h.Z
+  left := e.inv.right ≫ h.left
+  right := e.inv.right ≫ h.right
+  w := by
+    have := Arrow.mk_hom g ▸ Arrow.w_mk_right e.inv
+    simp_rw [← reassoc_of% this, h.w]
+  isLimit := Fork.isLimitOfIsos _ h.isLimit _
+    (Arrow.rightFunc.mapIso e) (Iso.refl _) (Arrow.leftFunc.mapIso e)
+
+/-- `IsRegularMono f` is the assertion that `f` is a regular monomorphism. -/
+abbrev IsRegularMono {X Y : C} (f : X ⟶ Y) : Prop := Nonempty (RegularMono f)
+
+variable (C) in
+/-- The `MorphismProperty C` satisfied by regular monomorphisms in `C`. -/
+def MorphismProperty.regularMono : MorphismProperty C := fun _ _ f => IsRegularMono f
+
+@[simp]
+theorem MorphismProperty.regularMono_iff (f : X ⟶ Y) :
+    (MorphismProperty.regularMono C) f ↔ IsRegularMono f :=
+  by rfl
+
+instance MorphismProperty.regularMono.containsIdentities :
+    (MorphismProperty.regularMono C).ContainsIdentities where
+  id_mem _ := ⟨RegularMono.ofIso <| Iso.refl _⟩
+
+instance MorphismProperty.regularMono.respectsIso :
+    (MorphismProperty.regularMono C).RespectsIso :=
+  RespectsIso.of_respects_arrow_iso _ (fun _ _ e h ↦ ⟨.ofArrowIso e (h := h.some)⟩)
+
+instance isRegularMono_of_regularMono (f : X ⟶ Y) [h : RegularMono f] : IsRegularMono f := ⟨h⟩
+
+instance (priority := low) regularMonoOfIsRegularMono (f : X ⟶ Y) [h : IsRegularMono f] :
```

Comment:
```suggestion
-- See note [lower instance priority]
instance (priority := low) regularMonoOfIsRegularMono (f : X ⟶ Y) [h : IsRegularMono f] :
```

Mathlib4's current setup to read library notes is a bit wonky, but the situation is expected to improve at some point, so I think it’s still useful to mention these notes when relevant.